twake/frontend/src/app/services/Apps/Messages/MessageReactionService.ts

```diff
@@ -0,0 +1,90 @@
+import { Message, ReactionType } from 'app/models/Message';
+import UserService from 'app/services/user/UserService';
+import MessageEditorManager from 'app/services/Apps/Messages/MessageEditorServiceFactory';
+import DepreciatedCollections from 'app/services/Depreciated/Collections/Collections.js';
+
+/**
+ * Service that allow you to manage message reactions
+ */
+class MessageReactionService {
+  collection: typeof DepreciatedCollections;
+
+  constructor() {
+    this.collection = DepreciatedCollections.get('messages');
+  }
+
+  /**
+   * Get user reactions in a message
+   * @param message Message
+   * @param userId string
+   */
+  public getUserReactions(message: Message, userId?: string): ReactionType[] {
+    const { reactions } = message;
+    const userReactions: ReactionType[] = [];
+
+    if (reactions?.length && userId) {
+      reactions.forEach(r => (r.users.includes(userId) ? userReactions.push(r) : false));
+    }
```

Comment:
What about `[].filter` instead of this `forEach`?

Reply:
Done